src/libfetchers/git-utils.cc

```diff
@@ -584,7 +584,16 @@ struct GitRepoImpl : GitRepo, std::enable_shared_from_this<GitRepoImpl>
                     "  ssh-ed25519-sk\n"
                     "  ssh-rsa",
                     k.type);
-            allowedSigners += "* " + k.type + " " + k.key + "\n";
+
+            std::string type;
+            if (k.type == "ssh-ecdsa-sk")
+                type = "sk-ecdsa-sha2-nistp256@openssh.com";
```

Comment:
If you haven't tested this one, I rather would leave it as a TODO for now until someone tested it.